File: check_stocks.js
const https = require('http');
const { TextDecoder } = require('util');
const td = new TextDecoder('gbk');

const stockMap = {
  'sh600352': { name: 'ZJLS', cost: 15.952, stop: 12.0 },
  'sz300033': { name: 'THS', cost: 423.488, stop: 280 },
  'sh831330': { name: 'PSDH', cost: 20.361, stop: 0 },
  'sz000988': { name: 'HGKJ', cost: 116.87, stop: 0 },
  'sh688295': { name: 'ZFSC', cost: 37.843, stop: 0 },
  'sh600487': { name: 'HTGD', cost: 42.391, stop: 0 },
  'sz300499': { name: 'GLGF', cost: 41.625, stop: 38.0 },
  'sh601168': { name: 'XBKY', cost: 24.863, stop: 0 },
  'sh600893': { name: 'HFDC', cost: 47.196, stop: 0 },
  'bj920046': { name: 'YNDL', cost: 329.555, stop: 27 },
  'bj430046': { name: 'SBR', cost: 0.478, stop: 0 },
  'sh600089': { name: 'TBDG', cost: 24.765, stop: 25.0 },
  'sh600114': { name: 'DMGF', cost: 32.428, stop: 25.0 },
  'sz301638': { name: 'NWSZ', cost: 32.635, stop: 28.0 },
};

const codes = Object.keys(stockMap).join(',');
const url = 'http://hq.sinajs.cn/list=' + codes;
https.get(url, { 'headers': { 'Referer': 'http://finance.sina.com.cn', 'User-Agent': 'Mozilla/5.0' } }, res => {
  let d = [];
  res.on('data', c => d.push(c));
  res.on('end', () => {
    const buf = Buffer.concat(d);
    const str = td.decode(buf);
    const lines = str.split('\n');
    let results = [];
    lines.forEach(l => {
      const m = l.match(/hq_str_(\w+)="([^"]+)"/);
      if (m) {
        const parts = m[2].split(',');
        const code = m[1];
        const info = stockMap[code];
        if (!info) return;
        const price = parseFloat(parts[3]);
        const yesterday = parseFloat(parts[2]);
        const chg = (price - yesterday).toFixed(3);
        const pct = ((price - yesterday) / yesterday * 100).toFixed(2);
        const pnl = ((price - info.cost) * (code.startsWith('bj') ? 1 : 1)).toFixed(0);
        const alert = (info.stop > 0 && price <= info.stop) ? ' STOP!' : '';
        console.log(info.name + '|' + price + '|' + chg + '|' + pct + '|' + pnl + '|' + alert);
      }
    });
  });
}).on('error', e => console.log(e));

File: check_stocks.py
import requests
import json

codes = ['sh600352', 'sz300033', 'sh600487', 'sh600893', 'sh601168', 'sh518880', 'sz430046', 'sh600114', 'sh600089']
names = {'sh600352':'浙江龙盛','sz300033':'同花顺','sh600487':'亨通光电','sh600893':'航发动力','sh601168':'西部矿业','sh518880':'黄金ETF','sz430046':'圣博润','sh600114':'东睦股份','sh600089':'特变电工'}

print('=== 持仓监控 09:30 ===')
for code in codes:
    try:
        url = f'https://qt.gtimg.cn/q={code}'
        r = requests.get(url, timeout=5)
        content = r.text
        # 格式: v_sh600352="...name~price~chg~chg_pct..."
        parts = content.split('~')
        if len(parts) > 10:
            price = float(parts[3])
            chg = float(parts[31]) if parts[31] else 0
            chg_pct = float(parts[32]) if parts[32] else 0
            name = names.get(code, parts[1])
            arrow = '↑' if chg_pct > 0 else '↓' if chg_pct < 0 else '-'
            print(f"{name}({code}) {price:.3f} {arrow}{abs(chg_pct):.2f}%")
        else:
            print(f"{code} 数据解析失败")
    except Exception as e:
        print(f"{code} 错误: {e}")
